lib/cartopy/mpl/geoaxes.py

```diff
@@ -261,10 +261,36 @@ def set_boundary(self, path, transform):
         self.set_transform(transform)
         self.stale = True
 
+    @contextlib.contextmanager
+    def _cx_disable_path_siplification(self, path):
+        """Contextmanager to temporarily disable path simplification."""
+        try:
+            should_simplify = path.should_simplify
+            path.should_simplify = False
+            yield
+        finally:
+            path.should_simplify = should_simplify
+
+    def _ensure_path_closed(self, path):
+        """Method to ensure that a path contains only closed sub-paths."""
+        # Split path into potential sub-paths and close all polygons
+        with self._cx_disable_path_siplification(path):
+            polygons = path.to_polygons()
+
+        m, l, c = mpath.Path.MOVETO, mpath.Path.LINETO, mpath.Path.CLOSEPOLY
+
+        codes, vertices = [], []
+        for poly in polygons:
+            vertices.extend([poly[0], *poly])
+            codes.extend([m, *[l]*(len(poly) - 1), c])
```

Comment:
I find the indirection with the new variables harder to follow here.
```suggestion
            codes.extend([mpath.Path.MOVETO,
                          *[mpath.Path.LINETO]*(len(poly) - 1),
                          mpath.Path.CLOSEPOLY])
```